lib/ramble/ramble/spack_runner.py

```diff
@@ -326,14 +327,51 @@ def add_include_file(self, include_file):
         if file_name in self._allowed_config_files:
             self.includes.append(include_file)
 
-    def concretize(self):
+    def copy_from_external_env(self, env_name_or_path):
         """
-        Concretize a spack environment.
+        Copy an external spack environment file into the generated environment.
 
-        This happens by generating a spack.yaml file to build
-        the packages that should be in the environment.
+        env_name_or_path can be either:
+         - Name of a named spack environment
+         - Path to an external spack environment
 
-        This command requires an active spack environment.
+        Returns:
+         - (bool) found_lock: True if a spack.lock file was copied. False otherwise.
+        """
+
+        self._check_active()
+
+        named_location_args = [
+            'location',
+            '-e',
+            env_name_or_path
+        ]
+
+        # Try to use a named environment first
+        try:
+            path = self.exe(*named_location_args, output=str).strip('\n')
```

Comment:
Should this be a `try` or more of a "check if we think it exists"? I don't think it matters too much either way, but I typically find it more graceful to check existence than generate an error

Reply:
I guess one option would be to flip it and check if the thing they gave us is indeed a file? What are the changes something is both a named env and a valid file?

Reply:
That's a good suggestion. I just updated this, let me know if you like that more.